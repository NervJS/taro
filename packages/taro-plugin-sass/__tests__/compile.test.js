const path = require('path')
const fs = require('fs')
const compileSass = require('../index')

const sassContent = `
.index {
  background-color: #ececec;
}
`

const indexPath = path.resolve(__dirname, '.', 'styles/index.scss')

const singlePath = path.resolve(__dirname, '.', 'styles/single.scss')
const singlePathContent = fs.readFileSync(singlePath)
const singleConfig = {
  resource: path.resolve(__dirname, '.', 'styles/variables.scss')
}
const singleConfigWithDirectory = {
  resource: 'styles/variables.scss',
  projectDirectory: path.resolve(__dirname, '.')
}

const multiplePath = path.resolve(__dirname, '.', 'styles/multiple.scss')
const multiplePathContent = fs.readFileSync(multiplePath)
const multipleConfig = {
  resource: [
    path.resolve(__dirname, '.', 'styles/variables.scss'),
    path.resolve(__dirname, '.', 'styles/mixins.scss')
  ]
}
const multipleConfigWithDirectory = {
  resource: [
    'styles/variables.scss',
    'styles/mixins.scss'
  ],
  projectDirectory: path.resolve(__dirname, '.')
}

describe('compileSass', () => {
  test('text compile only content', async () => {
    const res = await compileSass(sassContent, '', {})
    expect(res.css.toString()).toMatchSnapshot()
  })

  test('text compile only sass file path', async () => {
    const res = await compileSass('', indexPath, {})
    expect(res.css.toString()).toMatchSnapshot()
  })

  test('text compile sass file and single resource path', async () => {
    const res = await compileSass(singlePathContent, singlePath, singleConfig)
    expect(res.css.toString()).toMatchSnapshot()
  })

  test('text compile sass file and single resource file with directory', async () => {
    const res = await compileSass(singlePathContent, singlePath, singleConfigWithDirectory)
    expect(res.css.toString()).toMatchSnapshot()
  })

  test('text compile sass file and multiple resource path', async () => {
    const res = await compileSass(multiplePathContent, multiplePath, multipleConfig)
    expect(res.css.toString()).toMatchSnapshot()
  })

  test('text compile sass file and multiple resource file with directory', async () => {
    const res = await compileSass(multiplePathContent, multiplePath, multipleConfigWithDirectory)
    expect(res.css.toString()).toMatchSnapshot()
  })
})
