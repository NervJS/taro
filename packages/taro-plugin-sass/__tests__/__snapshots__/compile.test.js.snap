// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compileSass text compile only content 1`] = `
".index {
  background-color: #ececec; }
"
`;

exports[`compileSass text compile only sass file path 1`] = `
"body {
  font: 100% Helvetica, sans-serif;
  color: #333; }
"
`;

exports[`compileSass text compile sass file and multiple resource file with directory 1`] = `
".ui-color--info {
  color: #78A4FA; }

.ui-color--positive {
  color: #13CE66; }

.ui-color--negative {
  color: #FF4949; }

.ui-color--warning {
  color: #FFC82C; }

.box {
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg); }
"
`;

exports[`compileSass text compile sass file and multiple resource path 1`] = `
".ui-color--info {
  color: #78A4FA; }

.ui-color--positive {
  color: #13CE66; }

.ui-color--negative {
  color: #FF4949; }

.ui-color--warning {
  color: #FFC82C; }

.box {
  -webkit-transform: rotate(30deg);
  -ms-transform: rotate(30deg);
  transform: rotate(30deg); }
"
`;

exports[`compileSass text compile sass file and single resource file with directory 1`] = `
".index {
  font-size: 20px; }

.ui-color--info {
  color: #78A4FA; }

.ui-color--positive {
  color: #13CE66; }

.ui-color--negative {
  color: #FF4949; }

.ui-color--warning {
  color: #FFC82C; }
"
`;

exports[`compileSass text compile sass file and single resource path 1`] = `
".index {
  font-size: 20px; }

.ui-color--info {
  color: #78A4FA; }

.ui-color--positive {
  color: #13CE66; }

.ui-color--negative {
  color: #FF4949; }

.ui-color--warning {
  color: #FFC82C; }
"
`;
